v3io/dataplane/transport/httpclient.py

```diff
@@ -146,15 +197,18 @@ def _send_request_on_connection(self, request, connection):
             "Tx", connection=connection, method=request.method, path=path, headers=request.headers, body=request.body
         )
 
-        starting_offset = 0
         is_body_seekable = request.body and hasattr(request.body, "seek") and hasattr(request.body, "tell")
-        if is_body_seekable:
-            starting_offset = request.body.tell()
-
+        starting_offset = request.body.tell() if is_body_seekable else 0
         retries_left = self._request_max_retries
+        current_connection = connection  # Track the current connection for thread safety
```

Comment:
Maybe I'm missing something, but aliasing `connection` as `current_connection` doesn't seem to have any effect in this function.

Reply:
But the original connection variable is never used after aliasing.